server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -457,30 +462,37 @@ func (s *ByteStreamServer) handleAlreadyExists(ctx context.Context, stream bspb.
 	if err != nil {
 		return err
 	}
-	// Bazel 5.0.0 effectively requires that the committed_size match the total
-	// length of the *uploaded* payload. For compressed payloads, the uploaded
-	// payload length is not yet known, because it depends on the compression
-	// parameters used by the client. So we need to read the whole stream from the
-	// client in the compressed case.
-	//
-	// See https://github.com/bazelbuild/bazel/issues/14654 for context.
-	committedSize := r.GetDigest().GetSizeBytes()
-	if r.GetCompressor() != repb.Compressor_IDENTITY {
-		remainingSize := int64(0)
+	clientUploadedBytes := int64(len(firstRequest.Data))
+
+	ht := hit_tracker.NewHitTracker(ctx, s.env, false /*=ac*/)
```

Comment:
can we move the hit_tracker creation up in func (s *ByteStreamServer) Write() and then track when we call handleAlreadyExists?

Seems error prone to be creating hit trackers in two places (at two different levels) in the code.